packages/ubuntu_desktop_installer/lib/services/config_service.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:file/file.dart';
+import 'package:file/local.dart';
+import 'package:meta/meta.dart';
+import 'package:ubuntu_logger/ubuntu_logger.dart';
+
+/// @internal
+final log = Logger('config');
+
+class ConfigService {
+  ConfigService(this._path, {@visibleForTesting FileSystem? fs})
+      : _fs = fs ?? const LocalFileSystem();
+
+  final String _path;
+  final FileSystem _fs;
+
+  Future<Map<String, String?>> load() async {
+    final file = _fs.file(_path);
+    try {
+      final config =
+          await file.readAsLines().then((lines) => lines.toKeyValueMap());
+      log.debug('loaded ${config.length} entries from $_path');
+      return config;
+    } on FileSystemException catch (e) {
+      if (file.existsSync()) {
+        log.error('failed to load $_path', e);
+      }
+      return {};
+    }
+  }
+
+  Future<void> save(Map<String, String?> config) async {
+    try {
+      final file = _fs.file(_path);
+      final data = config.toKeyValueList().join('\n');
+      await file.create(recursive: true);
+      await file.writeAsString('$data\n');
+      log.debug('saved ${config.length} entries to $_path');
+    } on FileSystemException catch (e) {
+      log.error('failed to save $_path', e);
+    }
+  }
+}
+
+extension _KeyValueList on List<String> {
+  Map<String, String?> toKeyValueMap() {
+    return Map.fromEntries(
+      where((line) => !line.startsWith('#'))
+          .map((line) => line.split('='))
```

Comment:
`ini` additionally allows assignment with a colon (`:`). Is supporting that out of scope?

Reply:
I don't see any reason to support anything Bash-incompatible. The purpose is to transfer variables from the GUI to our own Bash scripts. :)